rackcity/views/asset_views.py

```diff
@@ -118,8 +120,23 @@ def asset_detail(request, id):
     """
     Retrieve a single asset.
     """
+    change_plan = None
+    serializer = None
+    if request.query_params.get("change_plan"):
+        (change_plan, response) = get_change_plan(request.query_params.get("change_plan"))
+        if response:
+            return response
     try:
-        asset = Asset.objects.get(id=id)
+        if change_plan:
+            assets, assets_cp = get_assets_for_cp(change_plan.id)
+            if assets_cp.filter(id=id).exists():
+                asset = assets_cp.get(id=id)
+            else:
+                asset = assets.get(id=id)
+            serializer = RecursiveAssetCPSerializer(asset)
```

Comment:
this should be serialized as RecursiveAssetSerializer instead right?

Reply:
lol good eye - i hadj ust fixed this when fixing the detailview bug